src/transformStateWithClones.js

```diff
@@ -8,6 +8,33 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  let stateCopy = { ...state };
+  const finalArray = [];
+
+  actions.forEach((action) => {
+    if (action.type === 'addProperties') {
+      stateCopy = Object.assign(stateCopy, action.extraData);
+      finalArray.push(stateCopy);
+    }
+
+    if (action.type === 'removeProperties') {
+      action.keysToRemove.forEach((key) => {
+        delete stateCopy[key];
+      });
+
+      finalArray.push(stateCopy);
+    }
+
+    if (action.type === 'clear') {
+      for (const property in stateCopy) {
+        delete stateCopy[property];
+      }
+
+      finalArray.push(stateCopy);
+    }
```

Comment:
If you have predetermined amount of options it is better to use switch/case structure instead
of multiple if/else blocks